### DISCUSSION TITLE
YT Issues

### DISCUSSION THREAD

**Discussion by Tomas622** [danielmiessler/Fabric#449] [Q&A]
Hey everyone, I'm on a mac and I just started with fabric but I'm having some issues with the yt stuff.
For one thing, what's the module that does the `yt --transcript` i've seen in examples? I did `pip3 install yt` and it did, but it said there was no `--transcript` option. I ended up getting a similar result with `youtube_transcript_api` but it always seems to cut off everything other than the last 2 or 3 minutes of the video, and I'd much rather use the one in the examples.

**Comment by Tomas622** [comment]
Still need help with this- I saw in another thread the module `youtube-dl` but that says it doesn't have a `--transcript` option? what's the `yt` module everyone's using? I'm probably missing something obvious to be honest but I'd love some advice

**Reply by deezplace** [reply]
In my README, in my install of Fabric, it mentions making the (yt ... --transcript) command by adding such to your bash. 
It's like an alias.
So try having a read of that =)

**Comment by black-backdoor** [comment]
`yt --transcript` is a function that comes with Fabric. You can find detailed information in the [README](https://github.com/danielmiessler/fabric?tab=readme-ov-file#helper-apps).

Make sure you have Fabric properly installed and configured. Note that `yt` uses Google's API, so ensure you have an API key set up. You can create one for free through the Google Developer Console.

**Reply by black-backdoor** [reply]
You need an youtube api key, you can create one for free. All you need is a Google account. Here's a helpful tutorial to get you started: [Tutorial Link](https://developers.google.com/youtube/v3/getting-started)

**Reply by Tomas622** [reply]
Ah ok so it's a fabric-specific thing, but how do I get it working in the first place? My Mac still says that it doesn't recognize `yt` as a command.

**Reply by EvocativeOlash** [reply]
> You need an youtube api key, you can create one for free. All you need is a Google account. Here's a helpful tutorial to get you started: [Tutorial Link](https://developers.google.com/youtube/v3/getting-started)

Yes the YouTube Data API v3

but It also asks for a google api key now, which one specifically do i need?

**Comment by fearby** [comment]
The YT function does not work for me despite installation ok and all APIs in place. Anyone fixed this?

**Comment by joelm-code** [comment]
1. Ensure Fabric Installation: Confirm that Fabric is properly installed and configured. Follow the setup instructions carefully to ensure all dependencies and configurations are in place.

Check for Updates: Make sure both Fabric and related modules are updated to the latest versions:
`pip install --upgrade youtube-dl fabric`


2. Wrap Command in Brackets: If the command contains special characters (like &), wrap the entire yt command in brackets to prevent it from detaching and running in the background:


`(yt --transcript "https://youtube.com/watch?v=uXs-zPc63kM") | fabric --stream --pattern extract_wisdom`

**Comment by deezplace** [comment]
from the README in my install

### Add aliases for all patterns

In order to add aliases for all your patterns and use them directly as commands ie. `summarize` instead of `fabric --pattern summarize`
You can add the following to your `.zshrc` or `.bashrc` file.

```bash
# Loop through all files in the ~/.config/fabric/patterns directory
for pattern_file in $HOME/.config/fabric/patterns/*; do
    # Get the base name of the file (i.e., remove the directory path)
    pattern_name=$(basename "$pattern_file")

    # Create an alias in the form: alias pattern_name="fabric --pattern pattern_name"
    alias_command="alias $pattern_name='fabric --pattern $pattern_name'"

    # Evaluate the alias command to add it to the current shell
    eval "$alias_command"
done

yt() {
    local video_link="$1"
    fabric -y "$video_link" --transcript
}
```
